### ISSUE TITLE
Unable to disable publishing to broker with pact-jvm-provider-junit5_2.12

### ISSUE THREAD

**Issue by erd** [DiUS/pact-jvm#799]
In our test setup, we want to control when verification results are uploaded to the broker. So, we set `pact.verifier.publishResults` to `false` for local development and set it to `true` on CI. However, it seems that when running the tests locally, the results are always uploaded to the broker.

build.gradle:
```
test {
    useJUnitPlatform()

    // Pact broker configuration
    systemProperty 'pactbroker.host', 'url'
    systemProperty 'pactbroker.port', '443'
    systemProperty 'pactbroker.protocol', 'https'
    systemProperty 'pactbroker.auth.username', System.getenv('PACT_BROKER_USERNAME')
    systemProperty 'pactbroker.auth.password', System.getenv('PACT_BROKER_PASSWORD')

    // Publish results back to the broker
    systemProperty 'pact.provider.version', project.version
    systemProperty 'pact.verifier.publishResults', 'false'
}

dependencies {
    testCompile 'au.com.dius:pact-jvm-provider-junit5_2.12:3.6.0-rc.1'
}
```

BootstrapContractTest.kt
```
@Provider("consumer-mobile-bff")
@PactBroker
@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@TestPropertySource(properties = ["url:http://localhost:9999"])
class BootstrapContractTest {
...
}
```

logs
```
2018-10-25 10:02:39,170 [DEBUG] [main] a.c.d.p.p.j.TestResultAccumulator - Received test result 'true' for Pact consumer-mobile-bff-consumer-android and Fetching application bootstrap information
2018-10-25 10:02:39,177 [DEBUG] [main] a.c.d.p.p.j.TestResultAccumulator - All interactions for Pact consumer-mobile-bff-consumer-android are verified
2018-10-25 10:02:45,228 [DEBUG] [main] a.c.d.p.p.HalClientBase - Posting JSON to https://url/pacts/provider/consumer-mobile-bff/consumer/consumer-android/pact-version/4a0c1337e0a6820802b01109e30f515121e4bbf5/verification-results
{"success":true,"providerApplicationVersion":"1.0.0"}
2018-10-25 10:02:46,220 [DEBUG] [main] a.c.d.p.p.HalClientBase - Got response HTTP/1.1 201 Created
2018-10-25 10:02:46,222 [DEBUG] [main] a.c.d.p.p.HalClientBase - Response body: {"providerName":"consumer-mobile-bff","providerApplicationVersion":"1.0.0","success":true,"verificationDate":"2018-10-25T17:02:46+00:00","_links":{"self":{"title":"Verification result","name":"Verification result 404 for Pact between consumer-android (v9.39.2) and consumer-mobile-bff","href":"url/pacts/provider/consumer-mobile-bff/consumer/consumer-android/pact-version/4a0c1337e0a6820802b01109e30f515121e4bbf5/verification-results/404"},"pb:pact-version":{"title":"Pact","name":"Pact between consumer-android (v9.39.2) and consumer-mobile-bff","href":"https://url/pacts/provider/consumer-mobile-bff/consumer/consumer-android/version/9.39.2"},"pb:triggered-webhooks":{"title":"Webhooks triggered by the publication of this verification result","href":"https://url/pacts/provider/consumer-mobile-bff/consumer/consumer-android/pact-version/4a0c1337e0a6820802b01109e30f515121e4bbf5/verification-results/404/triggered-webhooks"}}}
2018-10-25 10:02:46,233 [INFO] [main] a.c.d.p.p.ProviderVerifier - Published verification result of 'true' for consumer 'Consumer(name=consumer-android)'
```

I'm currently testing against 3.6.0-rc.1, but I've seen the same behavior with the last few releases.

**Comment by uglyog** [comment]
pact-jvm-provider-junit5 will now honour the `pact.verifier.publishResults` system property

**Comment by uglyog** [comment]
Released 3.5.24 with this fix

**Comment by erd** [comment]
I verified that the new release fixes the issue. Thanks so much for the quick fix!
